extensions/data-transfer/portability-data-transfer-microsoft/src/main/java/org/datatransferproject/transfer/microsoft/photos/MicrosoftPhotosImporter.java

```diff
@@ -246,6 +251,16 @@ static String cleanName(String name) {
         .collect(Collectors.joining(""));
   }
 
+  // Replace forbidden OneDrive file/directory characters with -
+  static String cleanOneDriveName(String name) {
```

Comment:
Yeah could you actually incorporate this logic into the transmogrification stuff I linked in the other comment? And make all the bits parameterizable, e.g.:
```
PhotoModel.cleanName(String forbiddenCharacters, String replaceWith, Int charLimit)
```

If you want to make an actual configuration object with defaults, etc. instead of using function params, that would also be cool.

Reply:
I added a TransmogrificationConfig class that can be used for all data types in the future, with noop defaults. Idea being each importer can extend it, define the fields, and thus use their own Transmogrification class.

Reply:
Awesome, thanks